clang/lib/Parse/ParseExprCXX.cpp

```diff
@@ -1385,6 +1385,11 @@ ExprResult Parser::ParseLambdaExpressionAfterIntroducer(
       Diag(RAngleLoc,
            diag::err_lambda_template_parameter_list_empty);
     } else {
+      // We increase the template depth before recursing into a requires-clause.
+      // The abbreviated generic lambdas thereof could have different template
+      // depths, avoiding substituting into wrong template parameters during the
```

Comment:
Can you point to where the use of `TemplateParameterDepth` happens? This makes sense but I am curious where this happens in this case.

Reply:
Done.